### ISSUE TITLE
Pasting images fails if clipboard also offers text/uri-list

### ISSUE THREAD

**Issue by WhyNotHugo** [flathub/org.signal.Signal#427]
When [my screenshot application](https://git.sr.ht/~whynothugo/shotman)  copies files into the clipboard selection, it offers two mime-types:

- `image/png` (where the accepting application gets a stream of bytes).
- `text/uri-list` (where the accepting application gets a path to a file).

It only `image/png` is offered, Signal will paste fine. But if both are offered Signal always attempts to paste reading the file. This fails due to Signal being sandboxed, and since it doesn't try the other mime-type, it just hands indefinitely due to a failure reading the file.

I'm not entirely sure how/if this can be fixed without tampering upstream, but since upstream indicates that Flathub-specific issues belong here, I figured it's best to start reporting this here.

**Comment by lbrunkho** [comment]
Signal probably doesn't have permission to access the location of the file in your clipboard. 

It looks like that even with the filesystem=host permission set for the Signal flatpak, this does NOT include other flatpak application directories in ~/.var/app/...

Example: I was using Reactions (a gif search application) and was unable to copy gifs into Signal. Pasting the link into a text editor revealed that the location of the image was in `/home/$USER/.var/app/dev.atoft.Reactions/cache/reactions.gif`. Giving the Signal flatpak access to `/home/$USER/.var/app/dev.atoft.Reactions/cache/` through Flatseal or similar and restarting Signal fixed this issue for me.

**Comment by WhyNotHugo** [comment]
> Signal probably doesn't have permission to access the location of the file in your clipboard.

Right, but there's also a data offer of type `image/png`. Signal could just prefer to read that one instead, and it will receive a file descriptor where it can read the image file.

**Comment by WhyNotHugo** [comment]
Can confirm that this is now fixed
